src/main/java/io/pravega/perf/PravegaWriterWorker.java

```diff
@@ -27,11 +27,20 @@
 
     final EventStreamWriter<byte[]> producer;
 
+    private final long writeWatermarkPeriodMillis;
+    private long nextNoteTime = System.currentTimeMillis();
```

Comment:
Is that guarded by a lock?

Reply:
A guard is not required as there is only one thread that uses it. I added a comment to indicate this.